v8-vtune-plugin code without error check

HreYang opened the pull request.

TianyouLi commented:
1.	Do not commit binary to source repo
2.	Do you commit your local settings like vscode project
3.	static __itt_domain* domain = NULL; that means only 1 domain allowed per process, right? If so, do you have error handling? Or close previous domain?
4.	To use other library, usually we do some function wrapper to avoid if library interface change.
5.	We should make sure the exception safe behavior, for example at destructor to close the domain.

HreYang commented:
Hi Tianyou,

Thank you so much for your comment.
And my reply is listed below.

Best regards,
Hre

From: Tianyou Li [mailto:notifications@github.com]
Sent: Saturday, May 25, 2019 10:04 AM
To: TianyouLi/v8-vtune-plugin <v8-vtune-plugin@noreply.github.com>
Cc: Yang, He <he.yang@intel.com>; Author <author@noreply.github.com>
Subject: Re: [TianyouLi/v8-vtune-plugin] v8-vtune-plugin code without error check (#1)


  1.  Do not commit binary to source repo
[Hre]: Sorry! I did not notice that at beginning. And in the latest version I removed the binary. And I sent a new pull request.
[cid:image002.jpg@01D512FB.F8A01670]

  1.  Do you commit your local settings like vscode project
[Hre]: Yes! Sorry! I did it at the beginning. But I changed it at latest version.

  1.  static __itt_domain* domain = NULL; that means only 1 domain allowed per process, right? If so, do you have error handling? Or close previous domain?
[Hre]: Per definition of the __itt_domain_create shown below, I defined this variable to be static. And it will not be destroyed in the whole execution time.
[cid:image003.png@01D512FA.1B854360]

  1.  To use other library, usually we do some function wrapper to avoid if library interface change.
[Hre]: Yes! Thank you for your suggestion.

  1.  We should make sure the exception safe behavior, for example at destructor to close the domain.
[Hre]: Sorry! But I could not get your point!
            Do you mean to wrapper the callee function which are from external library with try..catch mechamism?
Best regards,
Hre

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/TianyouLi/v8-vtune-plugin/pull/1?email_source=notifications&email_token=AIIIRMWDGZT3EIFBU6MG6YTPXCNBVA5CNFSM4HPOFWYKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWG3RPQ#issuecomment-495827134>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AIIIRMQKZTBXEQYNLN6H3BLPXCNBVANCNFSM4HPOFWYA>.

HreYang commented:
My comments inlined:

[Hre]: Per definition of the __itt_domain_create shown below, I defined this variable to be static. And it will not be destroyed in the whole execution time.
[Tianyou]  The domain api allowed to create multiple domains. So if we already have a domain, do we need to ‘end’ it before create a new one in the case we use global static? Or we should allowed to create multiple domain?

            Do you mean to wrapper the callee function which are from external library with try..catch mechamism?
[Tianyou]  We can combine the #4 and #5. We can design a VTuneDomain class, at its destructor, we can call ‘end’ for the domain, and we call also ‘free’ the string handlers if necessary which created by the domain, and we can do any cleanup. Remember the destructor will be called when our of block scope, when exception happens.

Regards,
Tianyou
Web Technology and Optimization<https://soco.intel.com/groups/wto>

From: Yang, He
Sent: Saturday, May 25, 2019 1:16 PM
To: TianyouLi/v8-vtune-plugin <reply@reply.github.com>; TianyouLi/v8-vtune-plugin <v8-vtune-plugin@noreply.github.com>; Li, Tianyou <tianyou.li@intel.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [TianyouLi/v8-vtune-plugin] v8-vtune-plugin code without error check (#1)

Hi Tianyou,

Thank you so much for your comment.
And my reply is listed below.

Best regards,
Hre

From: Tianyou Li [mailto:notifications@github.com]
Sent: Saturday, May 25, 2019 10:04 AM
To: TianyouLi/v8-vtune-plugin <v8-vtune-plugin@noreply.github.com<mailto:v8-vtune-plugin@noreply.github.com>>
Cc: Yang, He <he.yang@intel.com<mailto:he.yang@intel.com>>; Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [TianyouLi/v8-vtune-plugin] v8-vtune-plugin code without error check (#1)


  1.  Do not commit binary to source repo
[Hre]: Sorry! I did not notice that at beginning. And in the latest version I removed the binary. And I sent a new pull request.
[cid:image004.jpg@01D513AB.4BC4C620]

  1.  Do you commit your local settings like vscode project
[Hre]: Yes! Sorry! I did it at the beginning. But I changed it at latest version.

  1.  static __itt_domain* domain = NULL; that means only 1 domain allowed per process, right? If so, do you have error handling? Or close previous domain?
[Hre]: Per definition of the __itt_domain_create shown below, I defined this variable to be static. And it will not be destroyed in the whole execution time.
[cid:image005.png@01D513AB.4BC4C620]

  1.  To use other library, usually we do some function wrapper to avoid if library interface change.
[Hre]: Yes! Thank you for your suggestion.

  1.  We should make sure the exception safe behavior, for example at destructor to close the domain.
[Hre]: Sorry! But I could not get your point!
            Do you mean to wrapper the callee function which are from external library with try..catch mechamism?
Best regards,
Hre

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/TianyouLi/v8-vtune-plugin/pull/1?email_source=notifications&email_token=AIIIRMWDGZT3EIFBU6MG6YTPXCNBVA5CNFSM4HPOFWYKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWG3RPQ#issuecomment-495827134>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AIIIRMQKZTBXEQYNLN6H3BLPXCNBVANCNFSM4HPOFWYA>.
